test/test_http11.rb

```diff
@@ -183,4 +183,30 @@ def test_horrible_queries
     end
 
   end
+
+  def test_13632_workaround
+    pipes = []
+    threads = []
+    10.times do
+      r, w = IO.pipe
+      pipes << [r, w]
+      threads << Thread.new do
+        while r.gets
+        end rescue IOError
+        if Thread.current.pending_interrupt?
+          Thread.current.purge_interrupt_queue
```

Comment:
That would not be right, because of the reasons I've described above. However, I've improved a bit reporting there, to show a clear message in this case.